### PR TITLE
Simplified handling of params of GIS lookups.

### PR DESCRIPTION
**Pull request by sir-sigurd**

### CODE REVIEW

File: django/contrib/gis/db/models/lookups.py [hunk 1]
```diff
@@ -18,9 +18,21 @@ class GISLookup(Lookup):
...
+        self.process_rhs_params()
+
+    def process_rhs_params(self):
+        if self.rhs_params:
+            # Check if a band index was passed in the query argument.
+            if len(self.rhs_params) == (2 if self.lookup_name == 'relate' else 1):
+                self.process_band_indices()
+            elif len(self.rhs_params) > 1:
+                raise ValueError('Tuple too long for lookup %s.' % self.lookup_name)
+        elif isinstance(self.lhs, RasterBandTransform):
```
Comment (timgraham) [inline django/contrib/gis/db/models/lookups.py]: Any idea if this is needed per https://github.com/django/django/pull/8750?

File: django/contrib/gis/db/models/lookups.py [hunk 2]
```diff
@@ -302,28 +295,29 @@ class DistanceLookupBase(GISLookup):
...
 
-    def process_rhs(self, compiler, connection):
-        if not isinstance(self.rhs, (tuple, list)) or not 2 <= len(self.rhs) <= 4:
+    def process_rhs_params(self):
+        if not 1 <= len(self.rhs_params) <= 3:
             raise ValueError("2, 3, or 4-element tuple required for '%s' lookup." % self.lookup_name)
-        elif len(self.rhs) == 4 and not self.rhs[3] == 'spheroid':
-            raise ValueError("For 4-element tuples the last argument must be the 'speroid' directive.")
+        elif len(self.rhs_params) == 3 and self.rhs_params[2] != 'spheroid':
+            raise ValueError("For 4-element tuples the last argument must be the 'spheroid' directive.")
```
Comment (timgraham) [inline django/contrib/gis/db/models/lookups.py]: Could be nice to fix and test this typo separately.

File: django/contrib/gis/db/models/lookups.py [hunk 3]
```diff
@@ -302,28 +295,29 @@ class DistanceLookupBase(GISLookup):
...
 
         # Check if the second parameter is a band index.
-        if len(self.rhs) > 2 and not self.rhs[2] == 'spheroid':
+        if len(self.rhs_params) > 1 and not self.rhs_params[1] == 'spheroid':
             self.process_band_indices()
 
-        params = [connection.ops.Adapter(self.rhs[0])]
+    def process_rhs(self, compiler, connection):
+        params = [connection.ops.Adapter(self.rhs)]
```
Comment (timgraham) [inline django/contrib/gis/db/models/lookups.py]: I'd omit a blank line here.

File: django/contrib/gis/db/models/lookups.py (hunk 1 above)
Comment (sir-sigurd) [inline django/contrib/gis/db/models/lookups.py]: I think, yes. See #8782.

**Pull request merged**